### PR TITLE
(wip) support plural form

### PR DESCRIPTION
**Pull request by Djiit**
Didn't had that much time to put into it. Does it look like a correct way of doing it ?
I'd like to have a simpler way of providing a (model|field).pluralName everywhere. Open to any idea.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -4,18 +4,19 @@ datasource db {
...
-  exportRelationModifierClasses   = "true"
-  reExport                        = "true"
-  createDtoPrefix                 = "Create"
-  updateDtoPrefix                 = "Update"
-  dtoSuffix                       = "Dto"
-  entityPrefix                    = ""
-  entitySuffix                    = ""
+  provider                              = "node -r ts-node/register/transpile-only src/cli.ts"
+  output                                = "../src/@generated/prisma-generator-nestjs-dto"
+  outputToNestJsResourceStructure       = "true"
```
Comment (Djiit) [inline prisma/schema.prisma:9]: Needed this to test the PR. Should be reverted before merging.

File: src/generator/annotations.ts [hunk 2]
```diff
@@ -1,28 +1,29 @@
 export const DTO_IGNORE_MODEL = /@DtoIgnoreModel/;
 export const DTO_READ_ONLY = /@DtoReadOnly/;
+export const DTO_MODEL_NAME = /@DtoModelName (\w+)/;
 export const DTO_ENTITY_HIDDEN = /@DtoEntityHidden/;
 export const DTO_CREATE_OPTIONAL = /@DtoCreateOptional/;
 export const DTO_UPDATE_OPTIONAL = /@DtoUpdateOptional/;
 export const DTO_RELATION_REQUIRED = /@DtoRelationRequired/;
-export const DTO_RELATION_CAN_CRAEATE_ON_CREATE =
+export const DTO_RELATION_CAN_CREATE_ON_CREATE =
```
Comment (Djiit) [inline src/generator/annotations.ts:8]: Fixed this along the way. Can be extracted in another PR.

File: src/generator/index.ts [hunk 3]
```diff
@@ -50,21 +50,27 @@ export const run = ({
...
-          ? path.join(output, transformFileNameCase(model.name), 'dto')
-          : output,
-        entity: outputToNestJsResourceStructure
-          ? path.join(output, transformFileNameCase(model.name), 'entities')
-          : output,
-      },
-    }));
+    .map((model) => {
+      const modelName = getAnnotationValue(model, DTO_MODEL_NAME) || model.name;
+      model.plural = getAnnotationValue(model, DTO_MODEL_NAME) || model.name;
```
Comment (Djiit) [inline src/generator/index.ts:55]: Not sure this is the best way to add this attribute to every models.

**Comment by sebthom** [comment]
Dear Julien, thanks a lot for your interest in this project and your PR. We however stopped using this package ourselves and thus decided to suspend development until a new maintainer is found.
